src/components/app/data/utils.js

```diff
@@ -470,28 +472,45 @@ export function getAvailableCourseRuns({ course, isEnrollableBufferDays }) {
   if (!course?.courseRuns) {
     return [];
   }
-  const availableCourseRunsFilter = (courseRun) => {
-    if (!courseRun.isMarketable || isArchived(courseRun)) {
-      return false;
-    }
 
-    if (isEnrollableBufferDays === undefined) {
-      return courseRun.isEnrollable;
+  // These are the standard rules used for determining whether a run is "available".
+  const standardAvailableCourseRunsFilter = (courseRun) => (
+    courseRun.isMarketable && !isArchived(courseRun) && courseRun.isEnrollable
+  );
+
+  // These are more relaxed availability rules when late enrollment is applicable. We still never show archived courses,
+  // but the rules around the following fields are relaxed:
+  //
+  // * courseRun.isEnrollable: This field represents the enrollment window actually stored in the database. However,
+  //   during late enrollment we expand the end date of the enrollment window by isEnrollableBufferDays.
+  // * courseRun.isMarketable: This field is True when the run is published, has seats, and has a marketing URL. Since
+  //   late enrollment potentially means enrolling into an unpublished run, we must ignore the run state.
+  const lateEnrollmentAvailableCourseRunsFilter = (courseRun) => {
+    // If any of these early checks are True, then fail fast.
+    const baseNonAvailableChecks = [
+      isArchived(courseRun),
+      // The next two checks are in lieu of isMarketable which is otherwise overly sensitive to courserun state.
+      !('seats' in courseRun) || !courseRun.seats?.length,
+      !('marketingUrl' in courseRun) || !courseRun.marketingUrl,
```

Comment:
[nit] I believe `?` after `courseRun` should achieve the same results as `!('seats' in courseRun)`

``` jsx
const baseNonAvailalbeCheck = [
    isArchived(courseRun),
    !courseRun?.seats?.length,
    !courseRun?.marketingUrl,
]
```